src/cuckoocache.h

```diff
@@ -326,19 +329,25 @@ class cache
     }
 
     /** setup initializes the container to store no more than new_size
-     * elements.
+     * elements and no less than 2 elements.
      *
      * setup should only be called once.
      *
      * @param new_size the desired number of elements to store
-     * @returns the maximum number of elements storable
+     * @returns the maximum number of elements storable, or std::nullopt if
+     *          allocation failed
      */
-    uint32_t setup(uint32_t new_size)
+    std::optional<uint32_t> setup(uint32_t new_size)
     {
         // depth_limit must be at least one otherwise errors can occur.
-        depth_limit = static_cast<uint8_t>(std::log2(static_cast<float>(std::max((uint32_t)2, new_size))));
         size = std::max<uint32_t>(2, new_size);
-        table.resize(size);
+        depth_limit = static_cast<uint8_t>(std::log2(static_cast<float>(size)));
+        try {
+            table.reserve(size);
+            table.resize(size);
+        } catch (std::bad_alloc const&) {
+            return std::nullopt;
+        }
```

Comment:
This is never reached because there is already a global handler to handle bad allocs, so I think you can remove the `std::optional`.
https://github.com/bitcoin/bitcoin/blob/915c8ff123c770f0cce82358256ab02b93018767/src/init.cpp#L753-L764

Reply:
Oooof that sucks... There's no way to have a new handler that can access the `std::bad_alloc` object?

Reply:
AFAIK there doesn't seem to be a way :/

Reply:
Removed as of f66cd5b3aa, I think the existence of a new_handler makes this not too big of a problem now, and as was noted `-maxsigcachesize` is `DEBUG_ONLY`

Reply:
Closing, feel free to open if still a problem.